src/builder/QuarkBuilder.sol

```diff
@@ -501,17 +501,15 @@ contract QuarkBuilder {
         address withdrawer;
     }
 
-    // XXX support withdraw max
-    // XXX support Quotecall?
     function cometWithdraw(
         CometWithdrawIntent memory cometWithdrawIntent,
         Accounts.ChainAccounts[] memory chainAccountsList,
         PaymentInfo.Payment memory payment
     ) external pure returns (BuilderResult memory) {
         // XXX confirm that you actually have the amount to withdraw
 
-        // TODO: set this properly
-        bool useQuotecall = false;
+        bool isMaxWithdraw = cometWithdrawIntent.amount == type(uint256).max;
+        bool useQuotecall = payment.isToken && isMaxWithdraw;
```

Comment:
using Paycall for all max withdraws is easy.

I guess the more important part of this PR is establishing a structure for passing in information on a user's Comet positions (which will still be necessary for `cometWithdraw` since we need to know how much of the payment token the user will have after the operation has been executed).

Reply:
Agree that this can always be Paycall, so we can set this as `false` or remove the local variable altogether. I'd keep it for now just for readability and remove it if we run into stack too deep errors.

Reply:
updated to use Paycall